torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -358,28 +408,44 @@ class For : public StmtNode<For> {
   }
 
   For(const Var* var, const Expr* start, const Expr* stop, Stmt* body)
-      : var_(var), start_(start), stop_(stop), body_(body) {
-          CHECK(var && start && stop && body);
-      }
+      : var_(var), start_(start), stop_(stop) {
+    CHECK(var && start && stop && body);
+    Block* b = dynamic_cast<Block*>(body);
+    if (!b) {
+      b = new Block({body});
+    }
+    body_ = b;
+    set_parent(body_, this);
+  }
 
   For(const Var* var,
       const Expr* start,
       const Expr* stop,
       Stmt* body,
       const LoopOptions& loop_options)
-      : var_(var),
-        start_(start),
-        stop_(stop),
-        body_(body),
-        loop_options_(loop_options) {
-          CHECK(var && start && stop && body);
-        }
+      : var_(var), start_(start), stop_(stop), loop_options_(loop_options) {
+    CHECK(var && start && stop && body);
+    Block* b = dynamic_cast<Block*>(body);
+    if (!b) {
+      b = new Block({body});
+    }
+    body_ = b;
+    set_parent(body_, this);
```

Comment:
If most use of set_parent is set_parent(child, this), it might make sense to introduce

ClaimChild(child);